Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -104,18 +108,48 @@ void WebPermissionController::tryProcessingRequests()
         }
 
         currentRequest.isWaitingForReply = true;
-        m_page->sendWithAsyncReply(Messages::WebPageProxy::QueryPermission(currentRequest.origin, currentRequest.descriptor), [this, weakThis = WeakPtr { *this }](auto state, bool shouldCache) {
-            RefPtr protectedThis { weakThis.get() };
-            if (!protectedThis)
-                return;
 
+        // If the context is provided in the request, the API being queried is Notifications and so the permission
+        // will be obtained by using the context rather then sending a message to the UIProcess
+        if (currentRequest.context) {
             auto takenRequest = m_requests.takeFirst();
-            takenRequest.completionHandler(state);
-            if (shouldCache && state)
-                updateCache(takenRequest.origin, takenRequest.descriptor, *state);
-
-            tryProcessingRequests();
-        });
+            auto* client = takenRequest.context->notificationClient();
+
+            if (!client)
+                takenRequest.completionHandler(WebCore::PermissionState::Prompt);
+            else {
+                if (!takenRequest.context->isSecureContext())
+                    takenRequest.completionHandler(WebCore::PermissionState::Denied);
+                else {
+                    WebCore::NotificationPermission notiPer = client->checkPermission((takenRequest.context).get());
```

Comment:
WebKit style uses whole words. I think by "notiPer" you meant "notificationPermission".

Reply:
I see what you mean about the style here. Thank you for the feedback! The implementation for this pull request has now changed, but I'll keep this in mind for the future.